tcms/testcases/static/testcases/js/search.js

```diff
@@ -118,6 +118,37 @@ $(document).ready(function() {
 
     hookIntoPagination('#resultsTable', table);
 
+    const selectAllButton = $("#check-all")
+
+    selectAllButton.click(function () {
+        const rowCheckboxInputButton = $("input:checkbox[name='row-check']")
+        if (selectAllButton.prop("checked")) {
+            rowCheckboxInputButton.prop("checked", true);
+            table.rows().select()
+        } else {
+            rowCheckboxInputButton.prop("checked", false);
+            table.rows().deselect()
+        }
+    });
+
+    table.on('select', function (e, dt, type, indexes) {
+        if (type === 'row') {
+            const totalCheckBoxes = $("input:checkbox[name='row-check']").length;
+            const totalCheckedBoxes = $("input:checkbox[name='row-check']:checked").length;
+            if (totalCheckBoxes === totalCheckedBoxes) {
+                selectAllButton.prop("checked", true)
+            }
+        }
+    });
+
+    table.on('deselect', function (e, dt, type, indexes) {
+        if (type === 'row') {
+            if (selectAllButton.prop("checked")) {
+                selectAllButton.prop("checked", false)
```

Comment:
You can change the state unconditionally b/c once any of the rows is deselected the global state immediately becomes false.